benchmarking/example_classification.py

```diff
@@ -54,54 +66,45 @@
     X_test, y_test = X[test_idx], y[test_idx]
     X_valid, y_valid = X[valid_idx], y[valid_idx]
 
-    records = []
+    X_train_mols = [MolFromSmiles(smiles) for smiles in X_train]
+    X_valid_mols = [MolFromSmiles(smiles) for smiles in X_valid]
+    X_test_mols = [MolFromSmiles(smiles) for smiles in X_test]
 
-    fp_names = [
-        "ECFP",
-        "Atom Pairs",
-        "Topological Torsion",
-        "MACCS Keys",
-        "ERG",
-        "MAP4",
-        "MHFP",
-    ]
-    fprints = [
-        ECFPFingerprint,
-        AtomPairFingerprint,
-        TopologicalTorsionFingerprint,
-        MACCSKeysFingerprint,
-        ERGFingerprint,
-        MAP4Fingerprint,
-        MHFPFingerprint,
-    ]
-
-    clf_names = ["RF", "LogReg", "LGBM"]
-    classifiers = [
-        RandomForestClassifier,
-        LogisticRegression,
-        lgb.LGBMClassifier,
-    ]
-    classifier_kwargs = [
-        {"n_jobs": -1, "class_weight": "balanced"},
-        {
-            "class_weight": "balanced",
-            "penalty": None,
-            "n_jobs": -1,
-        },
-        {"n_jobs": -1, "class_weight": "balanced", "verbose": 0},
-    ]
+    records = []
 
     np.random.seed(42)
 
-    for fingerprint, fp_name in zip(fprints, fp_names):
+    for fingerprint in fingerprint_classes:
+        fp_name = fingerprint.__name__
         records.append({})
         records[-1]["fp_name"] = fp_name
-        print(fp_name, "Fingerprint")
+        print(fp_name)
         start = time()
         fp_transformer = fingerprint(n_jobs=-1)
-        X_fp_train = fp_transformer.transform(X_train)
-        X_fp_valid = fp_transformer.transform(X_valid)
-        X_fp_test = fp_transformer.transform(X_test)
+
+        try:
+            X_fp_train = fp_transformer.fit_transform(X_train)
+            X_fp_valid = fp_transformer.transform(X_valid)
+            X_fp_test = fp_transformer.transform(X_test)
+        except RuntimeError as e:
+            print(f" - Error: {e}")
+            records[-1]["error"] = str(e)
+            continue
+        except ValueError as e:
+            try:
+                conf_gen = ConformerGenerator()
+                X_train_conf = conf_gen.transform(X_train_mols)
+                X_valid_conf = conf_gen.transform(X_valid_mols)
+                X_test_conf = conf_gen.transform(X_test_mols)
+
+                X_fp_train = fp_transformer.fit_transform(X_train_conf)
+                X_fp_valid = fp_transformer.transform(X_valid_conf)
+                X_fp_test = fp_transformer.transform(X_test_conf)
+            except ValueError as e:
+                print(e.__class__.__name__)
+                print(f" - Error: {e}")
+                records[-1]["error"] = str(e)
+                continue
 
         scaler = MinMaxScaler()
         X_fp_train = scaler.fit_transform(X_fp_train)
```

Comment:
1. Scaling should be done after imputation
2. We can have extremely high values for some descriptors. While they can be clipped, this should be addressed here. Maybe use RobustScaler instead of MinMaxScaler?
3. Replace second `fit_transform` with `transform`

Reply:
Done